### ISSUE TITLE
documentation format

### ISSUE THREAD

**Issue by ht-thomas** [OSCA-Kampala-Chapter/autobot#20]
We're yet to the stage where we should start documenting the code for all those that are gonna use this project. Our documentation tools are going to be **mkdocs**, **mkdocstring** for extracting comments to turn them to documentation and **material-for-mkdocs** for the material design look of the documentation page. This [ article](https://realpython.com/python-project-documentation-with-mkdocs/) from realpython gives a brief on how we're going to do it.

**Comment by HassanBahati** [comment]
@ht-thomas is anyone working on this issue?

**Comment by ht-thomas** [comment]
@HassanBahati not really, I was hoping it would be a collective duty for everyone contributing to the code base to add docstring to the code they are submitting, in the format supported by mkdocstring (the tool we'll use to extract docstrings to be built into the overall documentation).

However with the way we're structuring the codebase, documentation can even be done independently of the code being submitted. So it's possible to work on documentation separately. If it's something you're interested in, you can gladly take it up.

**Comment by HassanBahati** [comment]
> @HassanBahati not really, I was hoping it would be a collective duty for everyone contributing to the code base to add docstring to the code they are submitting, in the format supported by mkdocstring (the tool we'll use to extract docstrings to be built into the overall documentation).
> 
> However with the way we're structuring the codebase, documentation can even be done independently of the code being submitted. So it's possible to work on documentation separately. If it's something you're interested in, you can gladly take it up.

Alright. I have submitted a pull request. Please review!

**Comment by reagan17** [comment]
is it possible to have like a google doc work space where  all those doc strings the developers add to their code can be extracted to eddited and latter combined to a documentation that can easily be transfered to github

**Comment by ht-thomas** [comment]
docstrings are necessary to be placed where they are to enable the automated generation of docs when the docs phase reaches
